From 979e90d7bc9ff6d9064e1f0378015abf40d86af2 Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Thu, 24 Dec 2020 18:47:56 +0800
Subject: [PATCH 2/2] media-libva: Add export AYUV color in
 vaExportSurfaceHandle when VA_EXPORT_SURFACE_COMPOSED_LAYERS is not set.

Example:
gst-launch-1.0 videotestsrc ! msdkvpp \
'video/x-raw(memory:DMABuf)',format=VUYA ! glimagesink

Upstream-Status: Submitted [https://github.com/intel/media-driver/pull/1115]

Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 media_driver/linux/common/ddi/media_libva.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/media_driver/linux/common/ddi/media_libva.cpp b/media_driver/linux/common/ddi/media_libva.cpp
index 778b7011..0e9f07d3 100755
--- a/media_driver/linux/common/ddi/media_libva.cpp
+++ b/media_driver/linux/common/ddi/media_libva.cpp
@@ -6679,6 +6679,8 @@ static uint32_t DdiMedia_GetDrmFormatOfSeparatePlane(uint32_t fourcc, int plane)
             return DRM_FORMAT_VYUY;
         case VA_FOURCC_UYVY:
             return DRM_FORMAT_UYVY;
+        case VA_FOURCC_AYUV:
+            return DRM_FORMAT_AYUV;
         case VA_FOURCC_Y210:
             return DRM_FORMAT_Y210;
         case VA_FOURCC_Y216:
-- 
2.17.1

